### PR TITLE
add support to junos_command module for netconf transport

### PR DESCRIPTION
**Pull request by privateip**
This adds support for the netconf transport and waitfor conditionals can
now express XML paths using the same . notation.

### CODE REVIEW

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

File: network/junos/junos_command.py [hunk 1]
```diff
@@ -20,15 +20,15 @@
...
+short_description: Run a list of commands on Juniper JUNOS devices
 description:
   - Netork devices running Juniper JUNOS can execute a variety of commands
     to display both configuration and state data.  This module will take
     an ordered set of commands and execute them in JUNOS and return the
-    command results.  It also supports adding conditionals the
-    argument set to force the module to meet a set of criteria before
-    proceeding.
+    command results.  It also supports adding a set of conditionals waitfor
+    statemetns that are evaluated and must be true before the task is
```
Comment (Qalthos) [inline network/junos/junos_command.py]: "statements" "considered"
There's something not quite right about the wording, too. Should it be "a set of conditional waitfor statements"?

File: network/junos/junos_command.py [hunk 2]
```diff
@@ -224,29 +138,31 @@ def main():
 
 
     commands = module.params['commands']
+    encoding = module.params['encoding'] or 'text'
+    if module.params['transport'] == 'netconf':
+        encoding = 'xml'
```
Comment (Qalthos) [inline network/junos/junos_command.py]: documentation above claims this is overridden only if not set. Perhaps

```
encoding = module.params.get('encoding')
if encoding is None:
    if module.params['transport'] == 'netconf':
        encoding = 'xml'
    else:
        encoding = 'text'
```

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
Thanks again to @privateip. This module is going into community review.

We notice that you are one of the maintainers (@privateip) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**